nemo/collections/nlp/modules/common/megatron/attention.py

```diff
@@ -503,7 +512,21 @@ def forward(
         if get_key_value:
             present = (key_layer, value_layer)
 
-        if checkpoint_core_attention:
+        if (
+            flash_attn_with_kvcache is not None
+            and self.use_flash_attention
+            and rotary_pos_emb is not None
+            and inference_max_sequence_len
```

Comment:
Should also check that we are not at the set_inference_key_value_memory stage?

Reply:
Yes! Done!